src/micro_http/src/common/headers.rs

```diff
@@ -138,43 +143,76 @@ impl Headers {
+                                    Err(RequestError::HeaderError(HttpHeaderError::InvalidValue(
+                                        entry[0].to_string(),
+                                        entry[1].to_string(),
+                                    )))
+                                }
                             }
                         }
                         Header::Accept => match MediaType::try_from(entry[1].trim().as_bytes()) {
                             Ok(accept_type) => {
                                 self.accept = accept_type;
                                 Ok(())
                             }
-                            Err(_) => Err(RequestError::UnsupportedHeader),
+                            Err(_) => {
+                                Err(RequestError::HeaderError(HttpHeaderError::InvalidValue(
+                                    entry[0].to_string(),
+                                    entry[1].to_string(),
+                                )))
+                            }
                         },
                         Header::TransferEncoding => match entry[1].trim() {
                             "chunked" => {
                                 self.chunked = true;
                                 Ok(())
                             }
-                            "identity; q=0" => Err(RequestError::InvalidHeader),
-                            _ => Err(RequestError::UnsupportedHeader),
+                            "identity; q=0" => Err(RequestError::HeaderError(
+                                HttpHeaderError::UnsupportedFeature(
+                                    entry[0].to_string(),
+                                    entry[1].to_string(),
+                                ),
+                            )),
+                            _ => Err(RequestError::HeaderError(HttpHeaderError::InvalidValue(
+                                entry[0].to_string(),
+                                entry[1].to_string(),
+                            ))),
                         },
                         Header::Expect => match entry[1].trim() {
                             "100-continue" => {
                                 self.expect = true;
                                 Ok(())
                             }
-                            _ => Err(RequestError::InvalidHeader),
+                            _ => Err(RequestError::HeaderError(
+                                HttpHeaderError::UnsupportedFeature(
+                                    entry[0].to_string(),
+                                    entry[1].to_string(),
+                                ),
+                            )),
                         },
                         Header::Server => Ok(()),
                     }
                 } else {
-                    Err(RequestError::UnsupportedHeader)
+                    Err(RequestError::HeaderError(HttpHeaderError::InvalidValue(
```

Comment:
Probably here we don't know if it's an invalid or an unsupported value. But we could go with unsupported to keep the previous behavior.